cocotb/scoreboard.py

```diff
@@ -66,6 +69,11 @@ def __init__(self, dut, reorder_depth=0, fail_immediately=True):  # FIXME: reord
         self.expected = {}
         self._imm = fail_immediately
 
+        warnings.warn(
+            "This Scoreboard implementation has been deprecated and will be removed soon.\n"
+            "If it works for you, please move the implementation into your own project and update your imports.",
```

Comment:
Maybe add the "follow cocotb's license agreement" part that you have for generators?